roles/kubernetes-apps/meta/main.yml

```diff
@@ -1,34 +1,46 @@
 ---
 dependencies:
   - role: kubernetes-apps/ansible
+    when:
+      - inventory_hostname == groups['kube-master'][0]
     tags:
       - apps
 
   - role: kubernetes-apps/efk
-    when: efk_enabled
+    when:
+      - efk_enabled
+      - inventory_hostname == groups['kube-master'][0]
     tags:
       - apps
       - efk
 
   - role: kubernetes-apps/helm
```

Comment:
Rather than restrict all the others role dependencies to `kube-master[0]`, would it be possible to leave the play bound use `delegate_to` and `loop` over `groups['kube-master']` here for just the helm dependency? I'm not certain if that can be done from a dependency in meta, but if so it would save us the changes to other dependencies here and in the playbooks.

Reply:
@chadswen I'm not sure to understand what you have in mind...
Current behaviour is to restrict the play to `groups['kube-master'][0]` (in `cluster.yml` and `update-cluster.yml`) and that's what I tried to stick to (except for helm of course).
Do you mean keeping `cluster.yml`, `update-cluster.yml` and `roles/kubernetes-apps/meta/main.yml` asis and loop over `groups['kube-master']` in `roles/kubernetes-apps/meta/main.yml` with `delegate_to: "{{ item }}"` only for helm ? 
If so, at first sight, every iteration will have facts from `groups['kube-master'][0]`, that could be a problem as every `when` condition including `inventory_hostname == groups['kube-master'][0]` would be true and the if condition to install/upgrade tiller would be tricky (I'm not sure loopindex is propagated this deep in meta dependencies.
Anyway, I'll give it a try tomorrow (can't test it from home :-/)

Reply:
Thanks, yes that is what I suggested. No big deal if it's a no go.

Reply:
@chadswen Not the way I thought but `with_items` and `delegate_to: "{{ item }}"` does not work in `roles/kubernetes-apps/meta/main.yml`:
`fatal: [node001]: FAILED! => {"failed": true, "msg": "'item' is undefined"}`